R/write_slides.R

```diff
@@ -0,0 +1,303 @@
+    if (str_detect(i, "/")) {
+      stop(
+        'Invalid filename. You included a forward slash in the file name. If you are trying to give a folder/directory location, use the `path =` argument.'
+      )
+    }
+    if (str_detect(i, "\\\\")) {
+      stop(
+        'Invalid filename. You included a backslash in the file name. If you are trying to give a folder/directory location, use the `path =` argument.'
+      )
+    }
+    if (!str_detect(i, '^[a-zA-Z0-9_-]+$')) {
+      stop(
+        'Invalid filename. Use only letters, numbers, hyphens, and underscores.'
+      )
+    }
+
+    # Remove .qmd if accidentally typed
+    i <- str_replace_all(i, '.qmd$', '')
+
+    # Validate filename: part 2
+    if (!is.character(i)) stop('Invalid filename: must be character.')
+    if (!str_detect(i, '^[a-zA-Z0-9_-]+$')) {
+      stop(
+        'Invalid filename. Use only letters, numbers, hyphens, and underscores.'
+      )
+    }
+  }
+
+  # Validation check for new_folder:
+  if (!is.character(new_folder)) stop('Invalid filename: must be character.')
+  if (!str_detect(new_folder, '^[a-zA-Z0-9_-]+$')) {
+    stop(
+      'Invalid "new_folder" argument. Use only letters, numbers, hyphens, and underscores.'
+    )
+  }
+
+  # Check the new_folder argument:
+  if (new_folder == "none") {
+    path <- path
+  } else if (new_folder == "slides") {
+    path <- glue::glue("{path}/slides")
+  } else if (new_folder %in% c("slides_", "slide_")) {
+    path <- glue::glue("{path}/slides_{filenames[1]}")
+  } else {
+    path <- glue::glue("{path}/{new_folder}")
+  }
+
+  # Create the slide deck directory:
+  if (!dir.exists(path)) dir.create(path)
+
+  # Normalize the path for consistency
+  path <- normalizePath(path, mustWork = TRUE)
+
+  if (file.access(path, mode = 2) != 0) {
+    stop(glue::glue(
+      'You do not have permission to write to the path location: {path}\nTry `rUM::write_quarto(filename = "", path = "")`'
+    ))
+  }
+
+  # Part 1: Determine the slides template type:
+  # list of valid slide templates
+  valid_templates <- c("miami", "rmed2025")
+
+  # 1a. Check if chosen template is an available template
+  if (template %in% valid_templates) {
+    # 1b. Check if using example themed template
+    if (example) {
+      slide_path <- glue::glue("gists/quarto_slides_example_{template}.qmd")
+    } else {
+      slide_path <- glue::glue("gists/quarto_slides_{template}.qmd")
+    }
+
+    # 2a. Check if template == "none"
+  } else if (template == "none") {
+    # 2b. Check if using example themed template
+    if (example) {
+      slide_path <- "slides/rUM_the_package.qmd"
+    } else {
+      slide_path <- "gists/quarto_slides.qmd"
+    }
+    
+    # 3. Supplied named template is not valid, so use generic template
+  } else {
+    slide_path <- "slides/rUM_the_word.qmd"
+  }
+
+  # Complete template_path:
+  template_path <- system.file(slide_path, package = "rUM")
+
+  # Part 2: Create each slide .qmd file:
```

Comment:
write_slides() should basically just be this part, and calls to all the valid args / helper functions

Reply:
No it needs the stuff above as part of the core